packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -2051,11 +2051,19 @@ def _schedule_round(self, round_cls: AppState) -> None:
 
         # self.state will point to last result,
         # or if not available to the initial state
-        last_result = (
-            self._round_results[-1]
-            if len(self._round_results) > 0
-            else self._initial_synchronized_data
-        )
+        if self._round_results:
+            synchronized_data_class = getattr(
+                round_cls, "synchronized_data_class", None
+            )
+            if not synchronized_data_class:
+                self.logger.warning(f"No `synchronized_data_class` set on {self}")
+                last_result = self._round_results[-1]
+            else:
+                synchronized_data_class = round_cls.synchronized_data_class  # type: ignore
+                last_result = synchronized_data_class(db=self._round_results[-1].db)
+                self._round_results[-1] = last_result
```

Comment:
why this assignment?

Reply:
so that in the next round you get the expected `SynchronizedData` type, if you assigned it as `synchronized_data_class`

Reply:
you are not allowed to manipulate this here. your other branches don't manipulate it either.

Reply:
yes I don't like this either. The alternative is that in each skills' BaseBehaviour (note: not all skills have a base behaviour defined) we cast it to the local `SynchronizedData` type. 
```
    @property
    def synchronized_data(self) -> SynchronizedData:
        """Return the synchronized data."""
        return cast(SynchronizedData, super().synchronized_data)
```
to
```
    @property
    def synchronized_data(self) -> SynchronizedData:
        """Return the synchronized data."""
        return SynchronizedData(db=super().synchronized_data.db)
```
but then the access via AbciApp will still return an instance of the previous type. I would also note that this is not a bug, but a feature, the bug is the `cast(SynchronizedData, super().synchronized_data)`